front/lib/api/resource_wrappers.ts

```diff
@@ -116,44 +198,149 @@ function withSpaceFromRoute<T, A extends SessionOrKeyAuthType>(
  * also supports the legacy usage of connectors with /w/[wId]/data_source/[dsId]/
  */
 function withDataSourceFromRoute<T, A extends SessionOrKeyAuthType>(
-  handler: ResourceHandler<T, "dataSource", A>
-) {
+  handler: ResourceHandler<T, A>
+): ResourceHandler<T, A> {
   return async (
     req: NextApiRequest,
     res: NextApiResponse<WithAPIErrorResponse<T>>,
     auth: Authenticator,
+    resources: Partial<ResourceMap<ResourceKey>>,
+    options: Partial<OptionsMap<ResourceKey>>,
     sessionOrKeyAuth: A
   ) => {
     const { dsId } = req.query;
-    const { wId } = req.query;
-    if (typeof dsId !== "string" || typeof wId !== "string") {
-      return apiError(req, res, {
-        status_code: 400,
-        api_error: {
-          type: "invalid_request_error",
-          message: "Invalid path parameters.",
-        },
-      });
-    }
 
-    const dataSource = await DataSourceResource.fetchById(auth, dsId);
+    if (dsId) {
+      const { wId } = req.query;
```

Comment:
This should rely on the authenticator no?

Reply:
Actually it looks unused so we should just remove that no?

Reply:
yes, it just checks the format of wId parameter, can definitely be removed from here

Reply:
Yep was already checked to construct the auth I presume